wafer/pages/models.py

```diff
@@ -161,14 +165,28 @@ def save(self, *args, **kwargs):
 
 def page_menus(root_menu):
     """Add page menus."""
-    for page in Page.objects.filter(include_in_menu=True):
-        path = page.get_path()
-        menu = path[0] if len(path) > 1 else None
-        try:
-            root_menu.add_item(page.name, page.get_absolute_url(), menu=menu)
-        except MenuError as e:
-            logger.error("Bad menu item %r for page with slug %r."
-                         % (e, page.slug))
+    for page in Page.objects.filter(include_in_menu=True, parent=None).prefetch_related("children"):
+        subpages = page.children.all()
+        if len(subpages) > 0:
+            root_menu.add_menu(
+                page.slug,
+                page.name,
+                [],
+                sort_key=page.menu_order,
+            )
+            for subpage in subpages:
+                root_menu.add_item(
+                    subpage.name,
+                    subpage.get_absolute_url(),
+                    menu=page.slug,
+                    sort_key=subpage.menu_order,
+                )
+        else:
+            root_menu.add_item(
+                page.name,
+                page.get_absolute_url(),
+                sort_key=page.menu_order,
+            )
```

Comment:
because the only case where a childpage is added is after the parent page was already added, so the case where a page is added to the wrong menu is guaranteed not to happen.